adala/skills/collection/label_studio.py

```diff
@@ -79,10 +80,11 @@ async def aapply(
                 instructions_template=self.instructions,
                 response_model=ResponseModel,
             )
-            ner_tag = self.has_ner_tag()
-            if ner_tag:
+            for ner_tag in self.ner_tags():
                 input_field_name = ner_tag.objects[0].value.lstrip('$')
                 output_field_name = ner_tag.name
                 quote_string_field_name = 'text'
-                output = extract_indices(pd.concat([input, output], axis=1), input_field_name, output_field_name, quote_string_field_name)
+                df = pd.concat([input, output], axis=1)
+                output = validate_output_format_for_ner_tag(df, input_field_name, output_field_name)
```

Comment:
Could we just add `validate_output_format_for_ner_tag` to the beginning of `extract_indices`? I figure we'd want this validation included in the calls to `extract_indices` in `EntityExtraction.apply` and `EntityExtraction.aaply` too, or is there too much overhead or something?

Reply:
That's a good point for backwards compat, I was thinking that `validate_output_format_for_ner_tag` is explicitly label-studio related and shouldn't be mixed up with generic NER functionality in the same function as extract_indices, but it should be used in the EntityExtraction skill too